### PR TITLE
Added support for underline and strikethrough style

### PR DESCRIPTION
**Pull request by Amir-P**

### CODE REVIEW

File: packages/zefyr/example/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -244,13 +244,15 @@
 				"${PODS_ROOT}/../Flutter/Flutter.framework",
```
Comment (cgestes) [inline packages/zefyr/example/ios/Runner.xcodeproj/project.pbxproj:244]: Are those changes requires?

File: packages/zefyr/example/ios/Podfile.lock [hunk 2]
```diff
@@ -4,6 +4,14 @@ PODS:
     - Flutter
   - image_picker (0.0.1):
     - Flutter
+  - path_provider (0.0.1):
+    - Flutter
+  - path_provider_linux (0.0.1):
```
Comment (cgestes) [inline packages/zefyr/example/ios/Podfile.lock:9]: Are the changes to this file required?

File: packages/zefyr/pubspec.yaml [hunk 3]
```diff
@@ -13,7 +13,9 @@ dependencies:
   collection: ^1.14.6
   url_launcher: ^5.0.0
   quill_delta: ^2.0.0
-  notus: ^1.0.0-dev
```
Comment (cgestes) [inline packages/zefyr/pubspec.yaml:16]: This needs to be dropped

File: packages/zefyr/lib/src/widgets/controller.dart [hunk 4]
```diff
@@ -173,6 +171,14 @@ class ZefyrController extends ChangeNotifier {
     super.dispose();
   }
 
+  /// List of style keys which can be toggled for insertion
+  List<String> get _insertionToggleableStyleKeys => [
```
Comment (cgestes) [inline packages/zefyr/lib/src/widgets/controller.dart:175]: Can you move this outside the class? Seems static.

**Review by cgestes — CHANGES_REQUESTED** [review]

**Comment by cgestes** [comment]
Btw, I am not the maintainer, just helping here :)

File: packages/zefyr/lib/src/widgets/controller.dart [hunk 5]
```diff
@@ -5,6 +5,14 @@ import 'package:notus/notus.dart';
 import 'package:quill_delta/quill_delta.dart';
 import 'package:zefyr/util.dart';
 
+/// List of style keys which can be toggled for insertion
+List<String> get _insertionToggleableStyleKeys => [
```
Comment (cgestes) [inline packages/zefyr/lib/src/widgets/controller.dart:9]: Can you turn it into a variable?

**Review by cgestes — COMMENTED** [review]
Looks good for me except the comment.

Let's wait for @pulyaevskiy review

**Comment by cgestes** [comment]
Also we may have to bump Notus version so that the test passes. I dont know how it works.

File: packages/zefyr/lib/src/widgets/controller.dart (hunk 5 above)
Comment (Amir-P) [reply packages/zefyr/lib/src/widgets/controller.dart]: Done 👍

**Comment by pulyaevskiy** [comment]
Thanks for taking time to submit this PR. I will take a look at it over this weekend.

File: packages/notus/lib/src/document/attributes.dart [hunk 6]
```diff
@@ -87,6 +91,12 @@ class NotusAttribute<T> implements NotusAttributeBuilder<T> {
   /// Italic style attribute.
   static const italic = _ItalicAttribute();
 
+  /// Underline style attribute.
+  static const underline = _UnderlineAttribute();
+
+  /// Strikethrough style attribute.
+  static const strike = _StrikeAttribute();
```
Comment (pulyaevskiy) [inline packages/notus/lib/src/document/attributes.dart:98]: I feel like it maybe better to name this `strikethrough` for consistency sake (as well as all other occurrences of this name here). WDYT?

**Review by pulyaevskiy — COMMENTED** [review]
This LGTM overall, just have a naming comment. PTAL.

Thanks again for taking time to work on this.

File: packages/notus/lib/src/document/attributes.dart (hunk 6 above)
Comment (Amir-P) [reply packages/notus/lib/src/document/attributes.dart]: Yes, you're right I've renamed it.

File: packages/zefyr/lib/src/widgets/theme.dart [hunk 7]
```diff
@@ -68,6 +68,12 @@ class ZefyrThemeData {
   /// Style of italic text.
   final TextStyle italic;
 
+  /// Style of underline text.
+  final TextStyle underline;
+
+  /// Style of strikethrough text.
+  final TextStyle strike;
```
Comment (pulyaevskiy) [inline packages/zefyr/lib/src/widgets/theme.dart:75]: Looks like this needs to be renamed as well.

Comment (Amir-P) [reply packages/zefyr/lib/src/widgets/theme.dart]: Totally forgot about that! It's Done.

**Review by pulyaevskiy — APPROVED** [review]
Nicely done, thanks again, LGTM now!

**Comment by lisadeng** [comment]
@cgestes @pulyaevskiy

**Comment by cgestes** [comment]
Hi,

cool for the features.

If you want to work on a local version of zefyr you can add this to your pubspec.yaml:

```
dependencies:
  zefyr:
    path: ../zefyr/packages/zefyr
```

You will have to amend the path to match your local checkout.

Formatting features may be straightforward.

The others features need proper separated issues to discuss how to do it :)

**Comment by lisadeng** [comment]
> Hi,
> 
> cool for the features.
> 
> If you want to work on a local version of zefyr you can add this to your pubspec.yaml:
> 
> ```
> dependencies:
>   zefyr:
>     path: ../zefyr/packages/zefyr
> ```
> 
> You will have to amend the path to match your local checkout.
> 
> Formatting features may be straightforward.
> 
> The others features need proper separated issues to discuss how to do it :)

Hi cgestes,
Noted with many thanks.

**Pull request merged**